### ISSUE TITLE
variables.ps1 fails on detecting InstallationInfoDirectory on the new 10.0.39 VMs

### ISSUE THREAD

**Issue by torbennehmer** [d365collaborative/d365fo.tools#829]
D365fo.tools fails on the newly released 10.0.39 VMs at least on the initial import when parsing variables.ps1, when detecting the installation info directory. The error originates from [Line 94ff on variables.ps1](https://github.com/d365collaborative/d365fo.tools/blob/f34367cc8843b494e5d8d7a8d0c9e3e2e47b9889/d365fo.tools/internal/scripts/variables.ps1#L89).

```powershell
$RegSplat = @{
    Path = "HKLM:\SOFTWARE\Microsoft\Dynamics\Deployment\"
    Name = "InstallationInfoDirectory"
}

$RegValue = $( if (Test-RegistryValue @RegSplat) { Join-Path (Get-ItemPropertyValue @RegSplat) "InstallationRecords" } else { "" } )
$Script:InstallationRecordsDir = $RegValue
```

On the 10.0.39 VM the registry key looks like this:

```text
InstallationInfoDirectory : F:\Deployment
PSPath                    : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Dynamics\Deployme
                            nt\
PSParentPath              : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Dynamics
PSChildName               : Deployment
PSDrive                   : HKLM
PSProvider                : Microsoft.PowerShell.Core\Registry
```

Drive F: is not available on the HyperV VM, so the subsequent execution fails with an unavailable drive error, excerpt from the debug loading:

![image](https://github.com/d365collaborative/d365fo.tools/assets/2812067/a1562582-a2ba-4b7f-b096-b90c25f96150)

My older VM used some folder on C: for this.

Apparently, the process continues, if you load the module a second time in a new shell, the error vanishes, the stored configuration seems to somehow hide the error on subsequent module loadings after the first one immediately after installation. It also has the subsequent side effect, that it won't load a module depending on d365fo.tools for the first time only. ~~Also, the error does not reappear after rebooting the machine.~~

When rebooting, you again need to try to import the module twice when used as a dependency, so the problem does persist over reboots, it does not fix itself in the configuration of d365fo.tools somehow.

I cannot say, if the information normally gathered from this folder is relevant or might interfere with d365fo.tools operations.

**Comment by torbennehmer** [comment]
Hi @FH-Inway thanks for the welcome, it's great that you're able to support this!
Thanks for the pointers, it is indeed frightening, what quality non-standards we see these days from MS, at least partly.
Maybe they rely on Copilot nowadays to create the images ... :-)
Anyway, the workaround with the Reg Patch works as intended, thanks for the quick response.

**Comment by FH-Inway** [comment]
Last week, Microsoft release a new version of the 10.0.39 VHD image that has the correct registry setting for `InstallationInfoDirectory`. I tested it and d365fo.tools can be imported on that version without an issue.

According to Yammer, there still remain issues related to that, but at least this particular issue should be resolved. I will close the issue for now, but feel free to add further comments or create a new issue.

![image](https://github.com/d365collaborative/d365fo.tools/assets/33372796/f69425a4-ef63-492f-b461-869876a7f608)
